src/Catalog.API/Catalog.API.csproj

```diff
@@ -21,6 +22,9 @@
   <!-- AI -->
   <ItemGroup>
     <PackageReference Include="Aspire.Azure.AI.OpenAI" />
+    <PackageReference Include="FastBertTokenizer" />
+    <PackageReference Include="Microsoft.SemanticKernel" />
+    <PackageReference Include="Microsoft.SemanticKernel.Connectors.Onnx" />
```

Comment:
@adityamandaleeka and I were talking a bit about this last week. Is the intention of eshop to "be samples and show all possible approaches"? Or is it to be more like a "real app" and only do what someone would do in a "real app"?

Would someone generally have multiple ways of tokenizing in their app switchable by configuration? Or would they "code to abstractions", but only depend on one tokenizer implementation? Is the thought here that I would use `FastBertTokenizer` in local development, and then swtich to Open AI in production?

Reply:
> Would someone generally have multiple ways of tokenizing in their app switchable by configuration?

It's not about the tokenizers, in fact I should be able to remove this FastBertTokenizer package reference; it's just used by the Onnx package.

> Is the thought here that I would use FastBertTokenizer in local development, and then swtich to Open AI in production?

It's about the embedding generator, and there are multiple reasons you might want to switch between them. As you say, you might want to use a local embedding generator in development and switch to a hosted service for production deployment / scale. You might want to be able to failover to a different generator if a given service is down. You might want to be able to do A/B testing between multiple generators. You might want to use a paid service up until you hit a certain budget and then fall back to a local model. And so on. You might not configure it exactly as I've done here (there are various ways you might have multiple embedding generators and then pick between them at run time), but I think what's here is a reasonable approximation.